core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -734,12 +711,43 @@ public PlanNode plan(Update node)
         return createMergePipeline(table, relationPlan, projectNode, rowIdSymbol, mergeRowSymbol);
     }
 
+    private PlanBuilder addCheckConstraints(Table table, PlanBuilder subPlanBuilder, RelationPlan targetTablePlan, Assignments.Builder assignments)
+    {
+        List<Expression> constraints = analysis.getCheckConstraints(table);
+        if (!constraints.isEmpty()) {
+            subPlanBuilder = subPlanBuilder.withNewRoot(new ProjectNode(
+                    idAllocator.getNextId(),
+                    subPlanBuilder.getRoot(),
+                    assignments.build()));
+
+            PlanBuilder constraintBuilder = subPlanBuilder.withScope(targetTablePlan.getScope(), targetTablePlan.getFieldMappings()).appendProjections(constraints, symbolAllocator, idAllocator);
+
+            List<Expression> predicates = new ArrayList<>();
+            for (Expression constraint : constraints) {
+                Expression symbol = constraintBuilder.translate(constraint).toSymbolReference();
+
+                Expression predicate = new IfExpression(
+                        // When predicate evaluates to UNKNOWN (e.g. NULL > 100), it should not violate the check constraint.
+                        new CoalesceExpression(coerceIfNecessary(analysis, symbol, symbol), TRUE_LITERAL),
+                        TRUE_LITERAL,
+                        new Cast(failFunction(plannerContext.getMetadata(), session, CONSTRAINT_VIOLATION, "Check constraint violation: " + constraint), toSqlType(BOOLEAN)));
+
+                predicates.add(predicate);
+            }
+
+            subPlanBuilder = subPlanBuilder.withNewRoot(new FilterNode(idAllocator.getNextId(), constraintBuilder.getRoot(), and(predicates)));
+        }
+        return subPlanBuilder;
+    }
+
     public MergeWriterNode plan(Merge merge)
     {
         MergeAnalysis mergeAnalysis = analysis.getMergeAnalysis().orElseThrow(() -> new IllegalArgumentException("analysis.getMergeAnalysis() isn't present"));
 
         List<List<ColumnHandle>> mergeCaseColumnsHandles = mergeAnalysis.getMergeCaseColumnHandles();
 
+        Table tagetTable = mergeAnalysis.getTargetTable();
```

Comment:
This variable is used from a single place (L857). Can we inline this? or I guess you wanted to use it from the following place. 
```java
FieldReference rowIdReference = analysis.getRowIdField(mergeAnalysis.getTargetTable());
```